matrix-event.c

```diff
@@ -50,5 +50,6 @@ void matrix_event_free(MatrixRoomEvent *event)
     g_free(event->txn_id);
     g_free(event->sender);
     g_free(event->event_type);
+    g_free(event->hook_data);
```

Comment:
It feels like a bit of a layering violation to make the event responsible for the ownership of the `hook_data` - shouldn't the event be treating the data as an opaque handle?

I can see that might get fiddly though. At least if you're going to do this, the header file needs some comments to say that the `hook_data` will be `g_freed`.